File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/components/gesturehandler/react/RNGestureHandlerEventDataExtractor.java
package versioned.host.exp.exponent.modules.api.components.gesturehandler.react;

import com.facebook.react.bridge.WritableMap;
import versioned.host.exp.exponent.modules.api.components.gesturehandler.GestureHandler;

public interface RNGestureHandlerEventDataExtractor<T extends GestureHandler> {
  void extractEventData(T handler, WritableMap eventData);
}

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/components/gesturehandler/react/RNGestureHandlerEventDataExtractor.kt
package versioned.host.exp.exponent.modules.api.components.gesturehandler.react

import com.facebook.react.bridge.WritableMap
import versioned.host.exp.exponent.modules.api.components.gesturehandler.GestureHandler

interface RNGestureHandlerEventDataExtractor<T : GestureHandler<T>> {
  fun extractEventData(handler: T, eventData: WritableMap)
}
